panel/io/document.py

```diff
@@ -175,6 +179,13 @@ async def _dispatch_msgs(doc, msgs):
     await asyncio.sleep(0.01)
     _dispatch_write_task(doc, _dispatch_msgs, doc, remaining)
 
+def _garbage_collect():
+    if (new_time:= time.monotonic()-_panel_last_cleanup) < GC_DEBOUNCE:
+        at = dt.datetime.now() + dt.timedelta(seconds=new_time)
+        state.schedule_task('gc.collect', _garbage_collect, at=at)
+        return
+    gc.collect()
```

Comment:
We can set `_panel_last_cleanup` here so that if a session is destroyed say every 4 seconds (maybe some automated polling script), then we don't end up *never* running cleanup!

That would also mean the first session destroyed will cause very quick gc but any immediately following session destruction will only call gc at 5s intervals.